ToolBox/FxcmDownloader/FxcmDataDownloader.cs

```diff
@@ -163,35 +161,37 @@ public IEnumerable<BaseData> Get(Symbol symbol, Resolution resolution, DateTime
             // initialize session
             RequestTradingSessionStatus();
 
-            Console.WriteLine("Downloading data from {0} to {1}...", startUtc.ToShortDateString(), endUtc.ToShortDateString());
+            Console.WriteLine("Downloading {0} data from {1} to {2}...", resolution, startUtc.ToString("yyyyMMdd HH:mm:ss"), endUtc.ToString("yyyyMMdd HH:mm:ss"));
+
+            //Find best FXCM  paramrs
+            IFXCMTimingInterval interval = ToFXCMInterval(resolution);
 
-            // download bars
-            var totalBars = new List<TradeBar>();
+            var totalTicks = (endUtc - startUtc).Ticks;
 
-            // calculate the maximum time span for one request (using 10-second bars)
-            const int maxBarsPerRequest = 300;
-            var timeSpanPerRequest = TimeSpan.FromSeconds(maxBarsPerRequest * 10);
+            // download data
+            var totalBaseData = new List<BaseData>();
 
-            var start = startUtc;
-            var end = startUtc + timeSpanPerRequest;
+            var end = endUtc;
 
-            // request loop
-            while (start < endUtc.AddDays(1))
+            do // 
             {
-                _currentBars.Clear();
+                //show progress
+                progressBar(Math.Abs((end - endUtc).Ticks), totalTicks, Console.WindowWidth / 2,'█');
```

Comment:
I love the idea of the progress bar, especially with FXCM :smile:

Perhaps we could generalize this and extend the concept to all downloaders by defining a `DownloadProgressChanged` event and moving the progress bar updates from the DataDownloader class to the event handler in the calling program (so in a WinForms app we could update a GUI progress bar).
This is just a note for a future PR, for now let's leave it as is.